contrib/pkg/createcluster/create.go

```diff
@@ -325,19 +333,71 @@ func (o *Options) getResourceHelper() (*resource.Helper, error) {
 func (o *Options) GenerateObjects() ([]runtime.Object, error) {
 	result := []runtime.Object{}
 
-	pullSecret, err := o.generatePullSecret()
+	pullSecretSecret, err := o.generatePullSecret()
+	if err != nil {
+		return nil, err
+	}
+
+	pullSecret, err := o.getPullSecret()
```

Comment:
Can we pass `pullSecret` in as a parameter to `generatePullSecret`? It looks strange to have `generatePullSecret` followed by `getPullSecret` with them having no visible relationship.

Reply:
will do